Added application file

LenyOstrich opened the pull request:
Added application file
Added link info property, update model, configuration class, proxy class
Updated tests to use proxy class
Added new methods to update or delete short links

LenyOstrich committed:
Added application file
Added link info property, update model, configuration class, proxy class
Updated tests to use proxy class
Added new methods to update or delete short links

panyukovnn commented on src/main/java/ru/iukr/linkshortener/model/LinkInfoUpdateModel.java:
```diff
@@ -0,0 +1,19 @@
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class LinkInfoUpdateModel {
```
Для чего в названии UpdateModel ?

panyukovnn replied:
Это по идее должен быть класс UpdateRequest и лежать в папке dto

LenyOstrich replied:
Fixed

panyukovnn commented on src/main/java/ru/iukr/linkshortener/repository/LinkInfoRepository.java:
```diff
@@ -12,4 +14,8 @@ public interface LinkInfoRepository {
+
+    void deleteLink(UUID uuid);
+
+    LinkInfo update(LinkInfoUpdateModel linkInfo);
```
Мне кажется не надо делать метод update в репозитории, вместо этого добавь пожалуйста метод .findById, при помощи него надо извлечь сущность, обновить поля и вызвать save()

LenyOstrich replied:
Fixed

panyukovnn commented on src/main/java/ru/iukr/linkshortener/repository/impl/LinkInfoRepositoryImpl.java:
```diff
@@ -28,4 +34,39 @@ public LinkInfo save(LinkInfo linkInfo) {
+                entryToRemove::add,
+                () -> log.info("Не удалось найти сущность по id: {}", uuid)
+        );
+        entryToRemove.forEach(entry -> storage.remove(entry.getKey()));
```
ситуация при которой идентификаторы продублируются в бд - недопустима, поэтому можно не вызывать удаление списка сущностй, достаточно сразу при обнаружении по id - удалить первую найденную сущность

LenyOstrich replied:
Fixed

LenyOstrich committed:
Renamed UpdateModel to UpdateRequest and moved to dto package
Removed update method from repository and added update logic to save method
Removed list from delete method

panyukovnn commented on src/main/java/ru/iukr/linkshortener/dto/UpdateRequest.java:
```diff
@@ -0,0 +1,19 @@
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class UpdateRequest {
```
Я чуть не так выразился ) надо назвать LinkInfoUpdateRequest, или UpdateLinkInfoRequest, как удобно, главное чтобы из названия было понятно, что для обновления именно LinkInfo

LenyOstrich replied:
Fixed

panyukovnn commented on src/main/java/ru/iukr/linkshortener/repository/impl/LinkInfoRepositoryImpl.java:
```diff
@@ -19,6 +23,8 @@ public Optional<LinkInfo> findByShortLink(String shortLink) {
 
     @Override
     public LinkInfo save(LinkInfo linkInfo) {
+        LinkInfo updatedLinkInfo = updateIfExists(linkInfo);
```
Кажется я другое поведение описал в предыдущих комментариях
Надо в LinkInfoService в методе updateLinkInfo сначала найти в репозитории нужную сущность с помощью метода .findById()
Затем обновить требуемые поля
Затем вызвать .save 

То поведение которое сейчас реализовано, на реальном проекте сложно будет применить, т.к. реализации репозиториев скрыты за фреймоврками, поэтому лучше через findById -> save

LenyOstrich replied:
Fixed

panyukovnn commented on src/main/java/ru/iukr/linkshortener/repository/impl/LinkInfoRepositoryImpl.java:
```diff
@@ -28,4 +34,33 @@ public LinkInfo save(LinkInfo linkInfo) {
+
+    @Override
+    public void deleteLink(UUID uuid) {
+        storage.entrySet().stream().filter(entry -> uuid == entry.getValue().getId()).findFirst().ifPresentOrElse(
```
UUID'ы следует сравнивать по .equals

Функциональные операторы лучше переносить на новые строки, для улучшения читабельности кода

LenyOstrich replied:
Fixed

LenyOstrich committed:
Renamed UpdateRequest
Removed update method from repository and added update logic to service

panyukovnn commented on src/main/java/ru/iukr/linkshortener/service/impl/LinkInfoServiceImpl.java:
```diff
@@ -47,6 +45,32 @@ public List<LinkInfoResponse> findByFilter() {
+            }
+            return toResponse(repository.save(linkToUpdate));
+        }
+        throw new NotFoundException("Не удалось найти сущность для обновления");
```
Я бы предложил выбрасывать исключение в самом начале, при получении пустого Optional

LinkInfo linkInfo = repository.findById(linkInfo.getId()).orElseThrow(() -> new NotFoundException("Не удалось найти сущность для обновления"));

LenyOstrich replied:
Fixed

LenyOstrich committed:
Code format refactoring
Fixing comments

panyukovnn commented on src/main/java/ru/iukr/linkshortener/repository/impl/LinkInfoRepositoryImpl.java:
```diff
@@ -28,4 +34,22 @@ public LinkInfo save(LinkInfo linkInfo) {
+    @Override
+    public void deleteLink(UUID uuid) {
+        storage.entrySet().stream()
+                .filter(entry -> uuid.equals(entry.getValue().getId()))
```
Здесь кстати можно вызывать метод findById, чтобы не дублировать код, но это уже не обязательно, все равно скоро заменим реализацию на бд
